services/ui_backend_service/api/card.py

```diff
@@ -133,20 +141,113 @@ async def get_card_content_by_hash(self, request):
         hash = request.match_info.get("hash")
         task = await self.get_task_by_request(request)
         if not task:
-            return web.Response(content_type="text/html", status=404, body="Task not found.")
+            return web.Response(
+                content_type="text/html", status=404, body="Task not found."
+            )
 
         cards = await get_cards_for_task(self.cache, task)
 
         if cards is None:
-            return web.Response(content_type="text/html", status=404, body="Card not found for task. Possibly still being processed.")
+            return web.Response(
+                content_type="text/html",
+                status=404,
+                body="Card not found for task. Possibly still being processed.",
+            )
 
         if cards and hash in cards:
             return web.Response(content_type="text/html", body=cards[hash]["html"])
         else:
-            return web.Response(content_type="text/html", status=404, body="Card not found for task.")
+            return web.Response(
+                content_type="text/html", status=404, body="Card not found for task."
+            )
+
+    @handle_exceptions
+    async def get_card_data_by_hash(self, request):
+        """
+        ---
+        description: Get the data of a card created for a task. Contains any additional updates needed by the card.
+        tags:
+        - Card
+        parameters:
+          - $ref: '#/definitions/Params/Path/flow_id'
+          - $ref: '#/definitions/Params/Path/run_number'
+          - $ref: '#/definitions/Params/Path/step_name'
+          - $ref: '#/definitions/Params/Path/task_id'
+          - $ref: '#/definitions/Params/Custom/invalidate'
```

Comment:
doesn't actually do anything with the invalidate query param

Reply:
Will Fix!. Bad copy pasta